salt/utils/cache.py

```diff
@@ -345,3 +348,32 @@ def context_cache_wrap(*args, **kwargs):
         return func(*args, **kwargs)
 
     return context_cache_wrap
+
+
+def verify_cache_version(cache_path):
+    """
+    Check that the cached version matches the Salt version.
+    If the cached version does not match the Salt version, wipe the cache.
+
+    :return: ``True`` if cache version matches, otherwise ``False``
+    """
+    if not os.path.isdir(cache_path):
+        os.makedirs(cache_path)
```

Comment:
Is this going to create the directory with the correct permissions?

Reply:
`os.mkdirs` is used all over gitfs so I believe this is the right way for gitfs.